clients/feeder/transaction.go

```diff
@@ -1,9 +1,49 @@
 package feeder
 
 import (
+	"errors"
+
 	"github.com/NethermindEth/juno/core/felt"
 )
 
+type ExecutionStatus uint8
+
+const (
+	Succeeded ExecutionStatus = iota + 1
+	Reverted
+)
+
+func (es *ExecutionStatus) UnmarshalJSON(data []byte) error {
+	switch string(data) {
+	case "SUCCEEDED":
+		*es = Succeeded
+	case "REVERTED":
+		*es = Reverted
+	default:
+		return errors.New("unknown ExecutionStatus")
+	}
+	return nil
+}
+
+type FinalityStatus uint8
+
+const (
+	AcceptedOnL2 FinalityStatus = iota + 1
+	AcceptedOnL1
```

Comment:
Do we need `Received` as well?

Reply:
Based on my understanding, no we do not. It's used internally by the feeder gateway for transactions that haven't been `REJECTED` or `ACCEPTED_*` yet. That's technically already used and we haven't needed it.